src/wallet/wallet2.cpp

```diff
@@ -3357,8 +3377,25 @@ bool wallet2::sign_tx(unsigned_tx_set &exported_txs, const std::string &signed_f
   {
     return false;
   }
-  LOG_PRINT_L3("Saving signed tx data: " << oss.str());
-  return epee::file_io_utils::save_string_to_file(signed_filename, std::string(SIGNED_TX_PREFIX) + oss.str());  
+  LOG_PRINT_L3("Saving signed tx data (with encryption): " << oss.str());
+  std::string ciphertext = encrypt_with_view_secret_key(oss.str());
+  if (!epee::file_io_utils::save_string_to_file(signed_filename, std::string(SIGNED_TX_PREFIX) + ciphertext))
+  {
+    LOG_PRINT_L0("Failed to save file to " << signed_filename);
+    return false;
+  }
+  // save signed raw tx without encryption
+  for (size_t i = 0; i < signed_txes.ptx.size(); ++i)
+  {
+    std::string tx_as_hex = epee::string_tools::buff_to_hex_nodelimer(tx_to_blob(signed_txes.ptx[i].tx));
+    std::string raw_filename = signed_filename + "_raw" + (signed_txes.ptx.size() == 1 ? "" : ("_" + std::to_string(i)));
+    if (!epee::file_io_utils::save_string_to_file(raw_filename, tx_as_hex))
+    {
+      LOG_PRINT_L0("Failed to save file to " << raw_filename);
+      return false;
+    }
+  }
+  return true;
```

Comment:
I think saving unencrypted is a bad idea unless the user wants it, since that's the whole point of encryption.

Reply:
OTOH, this is the local disk so I guess it's maybe alright...

Reply:
I see your point; the scenario I have in mind is that the user signs a tx, and copies the resulting file `signed_monero_tx` along with `signed_monero_tx_raw` to a USB stick, which then gets stolen etc. Now the thief can associate that particular tx with the user, which may be potentially problematic. Therefore, I'll make the saving of raw tx optional, activated only when `export` keyword is given in the CLI.